### PR TITLE
Add gha to upload Pyodide javascript bundle to R2

### PR DESCRIPTION
**Pull request by hoodmane**
This adds a github action that makes the Pyodide capnproto binary bundle and
uploads it to r2.

### CODE REVIEW

**Comment by dom96** [comment]
> Move the pyodide js code and this script to pyodide-build-scripts repo.

Should I wait for this to be done before reviewing?

**Review by dom96 — APPROVED** [review]
LGTM but are we still planning to move this to a different repo?

**Comment by hoodmane** [comment]
Yeah, eventually. We might as well merge this as-is and when we move repos we'll have to update how everything works.

File: .github/workflows/release-python-runtime.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      - name: Configure download mirrors
+        shell: bash
+        run: |
+          if [ ! -z "${{ secrets.WORKERS_MIRROR_URL }}" ] ; then
+            # Strip comment in front of WORKERS_MIRROR_URL, then substitute secret to use it.
+            sed -e '/WORKERS_MIRROR_URL/ { s@# *@@; s@WORKERS_MIRROR_URL@${{ secrets.WORKERS_MIRROR_URL }}@; }' -i.bak WORKSPACE
+          fi
+      - name: Bazel build
+        run: |
+          bazelisk build --config=release_linux //src/pyodide:pyodide.capnp.bin@rule
```
Comment (danlapid) [inline .github/workflows/release-python-runtime.yml:41]:
```suggestion
          bazel build --config=release_linux //src/pyodide:pyodide.capnp.bin@rule
```

Comment (danlapid) [inline .github/workflows/release-python-runtime.yml:41]: Might want to make use of the cache here and get this build line more similar to what we have in release.yml
`bazel build --disk_cache=~/bazel-disk-cache --strip=always --remote_cache=https://bazel:${{ secrets.BAZEL_CACHE_KEY }}@bazel-remote-cache.devprod.cloudflare.dev --config=ci --config=${{ matrix.bazel-config }} //src/workerd/server:workerd`

File: .github/workflows/release-python-runtime.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        with:
+          show-progress: false
+      - name: Setup Linux
+        if: runner.os == 'Linux'
+        run: |
+          export DEBIAN_FRONTEND=noninteractive
+          wget https://apt.llvm.org/llvm.sh
+          sed -i '/apt-get install/d' llvm.sh
+          chmod +x llvm.sh
+          sudo ./llvm.sh 15
```
Comment (danlapid) [inline .github/workflows/release-python-runtime.yml:28]:
```suggestion
          sudo ./llvm.sh 16
```

File: .github/workflows/release-python-runtime.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+          sudo apt-get install -y --no-install-recommends clang-15 lld-15 libunwind-15 libc++abi1-15 libc++1-15 libc++-15-dev
```
Comment (danlapid) [inline .github/workflows/release-python-runtime.yml:29]:
```suggestion
          sudo apt-get install -y --no-install-recommends clang-16 lld-16 libunwind-16 libc++abi1-16 libc++1-16 libc++-16-dev
```

File: .github/workflows/release-python-runtime.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+          echo "build:linux --action_env=CC=/usr/lib/llvm-15/bin/clang --action_env=CXX=/usr/lib/llvm-15/bin/clang++" >> .bazelrc
```
Comment (danlapid) [inline .github/workflows/release-python-runtime.yml:30]:
```suggestion
          echo "build:linux --action_env=CC=/usr/lib/llvm-16/bin/clang --action_env=CXX=/usr/lib/llvm-16/bin/clang++" >> .bazelrc
```

File: .github/workflows/release-python-runtime.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+          echo "build:linux --host_action_env=CC=/usr/lib/llvm-15/bin/clang --host_action_env=CXX=/usr/lib/llvm-15/bin/clang++" >> .bazelrc
```
Comment (danlapid) [inline .github/workflows/release-python-runtime.yml:31]:
```suggestion
          echo "build:linux --host_action_env=CC=/usr/lib/llvm-16/bin/clang --host_action_env=CXX=/usr/lib/llvm-16/bin/clang++" >> .bazelrc
```

**Comment by danlapid** [comment]
LGTM just a bit outdated since this PR was opened, let's update and merge

**Review by danlapid — APPROVED** [review]

**Pull request merged**